AsyncEnumerator.cs

```diff
@@ -12,6 +12,7 @@ public sealed class AsyncEnumerator<T> : IAsyncEnumerator<T>
         /// This exception is thrown when you call <see cref="Yield.Break"/>
         /// </summary>
         public sealed class AsyncEnumerationCanceledException : OperationCanceledException { }
+        public sealed class AsyncEnumerationDisposedException : OperationCanceledException { }
```

Comment:
I'm not even sure if this exception type is needed at all. Do you really care if enumeration is canceled because of a `CancellationToken` or if the enumerator is disposed?